File: qooxdoo-contrib/QWT/trunk/sushi/mork/src/main/java/org/qooxdoo/sushi/grammar/Symbol.java
/* ************************************************************************
   
   qooxdoo - the new era of web development
   
   http://qooxdoo.org
   
   Copyright:
     2006-2008 1&1 Internet AG, Germany, http://www.1and1.org
   
   License:
     LGPL: http://www.gnu.org/licenses/lgpl.html
     EPL: http://www.eclipse.org/org/documents/epl-v10.php
     See the LICENSE file in the project's top-level directory for details.
   
   Authors:
     * Michael Hartmeier (mlhartme)
   
 ************************************************************************ */

package org.qooxdoo.sushi.grammar;

/**
 * A stupid data container for Grammar.
 */

public class Symbol {
    /** productions for this symbol */
    public final int[] alternatives;

    /** productions using this symbol. */
    public final int[] users;

    /** ofsets in the using productions. */
    public final int[][] userOfs;

    public Symbol(int[] alternatives, int[] users, int[][] userOfs) {
        this.alternatives = alternatives;
        this.users = users;
        this.userOfs = userOfs;
    }
}

File: trunk/qooxdoo-contrib/SVG/trunk/source/class/svg/struct/Symbol.js
/* ************************************************************************

   Copyright:
     2010  Marc Puts

   License:
     LGPL: http://www.gnu.org/licenses/lgpl.html
     EPL: http://www.eclipse.org/org/documents/epl-v10.php
     See the LICENSE file in the project's top-level directory for details.

   Authors:
     * Marc Puts (marcputs)

************************************************************************ */

/**
 * Defines graphical template objects which can be instantiated by a {@link svg.Use} element.
 *
 * The use of Symbols for graphics that are used multiple times in the same document adds
 * struct and semantics.
 *
 * The key distinctions between a symbol and a {@link svg.Group} are:
 * <ul>
 *   <li>
 *     A symbol itself is not rendered. Only instances of a symbol element are rendered.
 *   </li>
 *   <li>
 *     A symbol can be scaled to fit within a rectangular viewport defined by
 *     the referencing Use element.
 *   </li>
 * </ul>
 *
 * More info:
 * <ul>
 *   <li>http://www.w3.org/TR/SVG/struct.html#SymbolElement</li>
 * </ul>
 */
qx.Class.define("svg.struct.Symbol",
{
  extend : svg.core.Element,
  
  include : [ svg.attributes.MViewBox,
              svg.attributes.MPreserveAspectRatio,
              svg.attributes.MStroke,
              svg.attributes.MTransform ],


  /**
   * @param styles {Map?null}
   *   Optional map of CSS styles, where the key is the name
   *   of the style and the value is the value to use.
   *
   * @param attributes {Map?null}
   *   Optional map of element attributes, where the key is
   *   the name of the attribute and the value is the value to use.
   */
  construct : function(styles, attributes) {
    this.base(arguments, "symbol", styles, attributes);
  },

  members : {}
});